pkg/instance/instance.go

```diff
@@ -406,122 +413,205 @@ func (s *Server) spdkInstanceLog(req *rpc.InstanceLogRequest, srv rpc.InstanceSe
 	return grpcstatus.Error(grpccodes.Unimplemented, "spdk instance log is not supported")
 }
 
+func (s *Server) handleNotify(ctx context.Context, notifyChan chan struct{}, srv rpc.InstanceService_InstanceWatchServer) error {
+	logrus.Info("Start handling notify")
+
+	for {
+		select {
+		case <-ctx.Done():
+			logrus.Info("Stopped handling notify")
+			return ctx.Err()
+		case <-notifyChan:
+			if err := srv.Send(&empty.Empty{}); err != nil {
+				return errors.Wrap(err, "failed to send instance response")
+			}
+		}
+	}
+}
+
 func (s *Server) InstanceWatch(req *empty.Empty, srv rpc.InstanceService_InstanceWatchServer) error {
 	logrus.Info("Start watching instances")
 
+	done := make(chan struct{})
+
+	clients := map[string]interface{}{}
+	go func() {
+		<-done
+
+		logrus.Info("Stopped clients")
+		for name, c := range clients {
+			switch c := c.(type) {
+			case *client.ProcessManagerClient:
+				c.Close()
+			case *spdkclient.SPDKClient:
+				c.Close()
+			}
+			delete(clients, name)
+		}
```

Comment:
Curious if we need to clean up clients here? I assume it's just a variable in the method stack, so it should be automatically free after being out from `InstanceWatch`?

Reply:
Yes, deleting it from a map is not necessary. Just make sure the clients only stores active clients, even it is not used anymore. WDYT?

Reply:
Sounds good, it's always good to ensure that. However, just clarified that everything is created in a stack memory, so they should be freed eventually after being out from it.

Reply:
>  I assume it's just a variable in the method stack, so it should be automatically free after being out from InstanceWatch?

Yeah, golang can automatically free the unused stack (garbage collection).